### PR TITLE
feat: upgrade assemblyscript faas demo(#256)

### PR DESCRIPTION
**Pull request by nanjingboy**
**What this PR does**:

Upgrade the wasm demo developed by AssemblyScript(#256)

**Special notes for your reviewer**:

The Node.js version should >=16.14.0

### CODE REVIEW

**Comment by nanjingboy** [comment]
@seeflood

**Review by zu1k — CHANGES_REQUESTED** [review]

File: demo/faas/code/assemblyscript/assembly/client.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    super(context_id, root_context);
+  }
+
+  onRequestBody(body_buffer_length: usize, _end_of_stream: bool): FilterDataStatusValues {
+    let body =  String.UTF8.decode(get_buffer_bytes(BufferTypeValues.HttpRequestBody, 0, body_buffer_length as u32));
+    if (body.indexOf("name=") === -1) {
+      log(LogLevelValues.error, "Param 'name' not found");
+    } else {
+      const name = body.slice("name=".length);
+      set_http_response_body(`There are ${String.UTF8.decode(invokeService("id_2", "", name))} inventories for ${name}.\n`);
```
[the remark this answers is not in the document]
Comment (nanjingboy) [reply demo/faas/code/assemblyscript/assembly/client.ts]: updated

**Review by nobodyiam — APPROVED** [review]
LGTM

**Comment by zu1k** [comment]
I'm not familiar with assemblyscript, but LGTM.

**Review by zu1k — APPROVED** [review]

**Comment by seeflood** [comment]
Thanks for your contribution! We can merge this PR and #475 after v0.4 released

**Review by seeflood — APPROVED** [review]
lgtm

**Review by zhenjunMa — APPROVED** [review]

**Pull request merged**